### PR TITLE
vcpkg update 2.4

### PR DESCRIPTION
**Pull request by daschuer**
This updates to the latest vcpkg environment. 
This uses the new release prefix for the release only triplets like upstream.

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
On Windows 11 the following 3 cases were tested successful:
x64_portable from release-only buildenv
x64_portable from full buildenv
x64_off from full buildenv

Due to the update of the libdjinterop VCPKG port to the required version, I got faster build times than ever:
Debug build (x64_off):  85s -> 66s
RelWithDebInfo: 115s -> 88s
(Full rebuild time of mixxx without mixxx-test)

**Comment by JoergAtGithub** [comment]
@fwcd or @m0dB 
Could one of you test this PR on Macintosh too.
Especially the case with the renamed triplet in the release-only buildenv archive. Does it still build for both supported CPU architectures?

**Review by JoergAtGithub — APPROVED** [review]

**Comment by fwcd** [comment]
Is there a reason why CI doesn't run on these PRs?

**Comment by fwcd** [comment]
In that case, I'd say it's fine. While it would of course be nice to test the arm64-native build too, the x86-64 native and cross-builds from CI should be enough to catch anything major (for now).

**Comment by fwcd** [comment]
Ah sorry, for some reason I thought this was a PR in `vcpkg`, that makes sense. I'll have a look

**Comment by fwcd** [comment]
```
$ arch -x86_64 cmake -DCMAKE_TOOLCHAIN_FILE=/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/scripts/buildsystems/vcpkg.cmake -B build
-- CMAKE_VERSION: 3.27.7
-- Using MIXXX_VCPKG_ROOT: /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c
-- Setting CMAKE_BUILD_TYPE to 'Debug' as none was specified.
-- The C compiler identification is AppleClang 15.0.0.15000040
-- The CXX compiler identification is AppleClang 15.0.0.15000040
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- VCPKG_TARGET_TRIPLET dir not found: /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012 Make sure the VCPKG build environment is installed and contains the build for the selected triplet.
CMake Error at CMakeLists.txt:59 (message):
  Did you download the Mixxx build environment using
  ´/Users/<user>/git/mixxx/tools/macos_buildenv.bat´
Call Stack (most recent call first):
  CMakeLists.txt:211 (FATAL_ERROR_MISSING_ENV)


-- Configuring incomplete, errors occurred!
```

We'll probably have to update CMakeLists to default to the right `VCPKG_TARGET_TRIPLET`

**Comment by fwcd** [comment]
This seems to be an issue that only happens when using `BUILDENV_RELEASE`

**Comment by fwcd** [comment]
I could see how this could become a bit annoying to check for in the CMakeLists, since release-mode dependencies do not necessarily imply building with `CMAKE_BUILD_TYPE != Debug` and we want to support all combinations of {release,debug} dependencies and {release,debug} Mixxx...

**Comment by fwcd** [comment]
Side note: `macos_release_buildenv.sh` is currently broken, since that script currently tries to execute `macos_buildenv.sh` instead of sourcing it.

File: tools/macos_buildenv.sh [hunk 1]
```diff
@@ -22,22 +22,22 @@ THIS_SCRIPT_NAME=${BASH_SOURCE[0]}
 if [ -n "${BUILDENV_ARM64}" ]; then
     if [ -n "${BUILDENV_RELEASE}" ]; then
         BUILDENV_BRANCH="2.4-rel"
-        BUILDENV_NAME="mixxx-deps-rel-2.4-arm64-osx-min1100-3e909e2"
-        BUILDENV_SHA256="f6f84f552e76f8e1e6617b22476da4642ce2fbcbc1e4eb593ad6b0aeb05ae206"
+        BUILDENV_NAME="mixxx-deps-rel-2.4-arm64-osx-min1100-release-c82358c"
```
Comment (fwcd) [inline tools/macos_buildenv.sh:25]: Another side note: Perhaps we could shorten `mixxx-deps-rel` to just `mixxx-deps` if we already encode the release mode into the triplet?

**Comment by fwcd** [comment]
Here's a preliminary patch:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d232f3707..e5faabb413 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -103,8 +103,12 @@ if(DEFINED MIXXX_VCPKG_ROOT)
   endif()
 endif()
 
-if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
-  set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}")
+if(NOT DEFINED VCPKG_TARGET_TRIPLET)
+  if(DEFINED ENV{VCPKG_TARGET_TRIPLET})
+    set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_TARGET_TRIPLET}")
+  elseif(DEFINED ENV{VCPKG_DEFAULT_TRIPLET})
+    set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}")
+  endif()
 endif()
 set(X_VCPKG_APPLOCAL_DEPS_INSTALL ON CACHE BOOL "Automatically copy dependencies into the install target directory for executables." FORCE)
 
diff --git a/tools/macos_buildenv.sh b/tools/macos_buildenv.sh
index 13ea89569b..36b4e76614 100755
--- a/tools/macos_buildenv.sh
+++ b/tools/macos_buildenv.sh
@@ -21,22 +21,26 @@ THIS_SCRIPT_NAME=${BASH_SOURCE[0]}
 
 if [ -n "${BUILDENV_ARM64}" ]; then
     if [ -n "${BUILDENV_RELEASE}" ]; then
+        VCPKG_TARGET_TRIPLET="arm64-osx-min1100-release"
         BUILDENV_BRANCH="2.4-rel"
-        BUILDENV_NAME="mixxx-deps-rel-2.4-arm64-osx-min1100-release-c82358c"
+        BUILDENV_NAME="mixxx-deps-rel-2.4-$VCPKG_TARGET_TRIPLET-c82358c"
         BUILDENV_SHA256="50dceb835f62ecb2013fdcf918a5793affde9334270e027c8a2a2f4b85861b42"
     else
+        VCPKG_TARGET_TRIPLET="arm64-osx-min1100"
         BUILDENV_BRANCH="2.4"
-        BUILDENV_NAME="mixxx-deps-2.4-arm64-osx-min1100-250db0f"
+        BUILDENV_NAME="mixxx-deps-2.4-$VCPKG_TARGET_TRIPLET-250db0f"
         BUILDENV_SHA256="7741284e77de9758b34bf55fcf105bb225b8df4e04eaf3470adea2d270526882"
     fi
 else
     if [ -n "${BUILDENV_RELEASE}" ]; then
+        VCPKG_TARGET_TRIPLET="x64-osx-min1012-release"
         BUILDENV_BRANCH="2.4-rel"
-        BUILDENV_NAME="mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c"
+        BUILDENV_NAME="mixxx-deps-rel-2.4-$VCPKG_TARGET_TRIPLET-c82358c"
         BUILDENV_SHA256="040e57362929283dbec161c46e92f6a6ea0a9dccaecab2a168da1057fdefeb9d"
     else
+        VCPKG_TARGET_TRIPLET="x64-osx-min1012"
         BUILDENV_BRANCH="2.4"
-        BUILDENV_NAME="mixxx-deps-2.4-x64-osx-min1012-250db0f"
+        BUILDENV_NAME="mixxx-deps-2.4-$VCPKG_TARGET_TRIPLET-250db0f"
         BUILDENV_SHA256="ef598265bd8c3a176b7d01e4d734d0c6c3f443a2957a26fe53826220f7bb8550"
     fi
 fi
@@ -86,10 +90,12 @@ case "$1" in
 
         export MIXXX_VCPKG_ROOT="${BUILDENV_PATH}"
         export CMAKE_GENERATOR=Ninja
+        export VCPKG_TARGET_TRIPLET="${VCPKG_TARGET_TRIPLET}"
 
         echo_exported_variables() {
             echo "MIXXX_VCPKG_ROOT=${MIXXX_VCPKG_ROOT}"
             echo "CMAKE_GENERATOR=${CMAKE_GENERATOR}"
+            echo "VCPKG_TARGET_TRIPLET=${VCPKG_TARGET_TRIPLET}"
         }
 
         if [ -n "${GITHUB_ENV}" ]; then
```

We have to make sure `VCPKG_TARGET_TRIPLET` is exported and make sure the `CMakeLists` actually respect that environment variable. Interesting that it worked on Windows though, I assume that is because the buildenv exports the variables to a CMakeSettings.json rather than just the environment?

**Comment by fwcd** [comment]
Tried building under Rosetta with

```
$ arch -x86_64 cmake -DCMAKE_TOOLCHAIN_FILE=/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/scripts/buildsystems/vcpkg.cmake /Users/<user>/git/mixxx -B build
$ arch -x86_64 cmake --build build
```

and got

```
[769/885] Linking CXX static library libmixxx-lib.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libmixxx-lib.a(moc_included_test.cpp.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libmixxx-lib.a(moc_included_test.cpp.o) has no symbols
[778/885] Linking CXX executable mixxx
FAILED: mixxx 
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -fdiagnostics-color=auto -g -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=10.12 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/mixxx.dir/src/main.cpp.o CMakeFiles/mixxx.dir/mixxx_autogen/PNK5WDWK6L/qrc_mixxx.cpp.o -o mixxx  libmixxx-lib.a  libmixxx-gitinfostore.a  -weak_framework iTunesLibrary  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libchromaprint.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libdjinterop.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libsqlite3.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libebur128.a  libfidlib.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libkeyfinder.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libFLAC.a  libFpClassify.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libmp3lame.a  libKaitai.a  libMP3GuessEnc.a  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/OpenGL.framework/OpenGL.tbd  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libvorbisfile.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libportaudio.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libjack.a  libPortAudioRingBuffer.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libportmidi.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libprotobuf-lite.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Concurrent.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5OpenGL.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5PrintSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5QuickWidgets.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Quick.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Qml.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Network.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Sql.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Svg.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Test.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Widgets.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Gui.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Xml.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/platforms/libqoffscreen.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/platforms/libqminimal.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/imageformats/libqgif.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/imageformats/libqico.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/imageformats/libqjpeg.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/imageformats/libqsvg.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/sqldrivers/libqsqlite.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/platforms/libqcocoa.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/plugins/styles/libqmacstyle.a  -weak_framework Accelerate  -weak_framework AudioToolbox  -weak_framework AudioUnit  -weak_framework AVFoundation  -weak_framework CoreAudio  -weak_framework CoreImage  -weak_framework CoreMedia  -weak_framework CoreMidi  -weak_framework IOSurface  -weak_framework VideoToolbox  libQueenMaryDsp.a  libReplayGain.a  libReverb.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/librubberband.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libfftw3.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libsamplerate.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libsndfile.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libSoundTouch.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libtag.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libavcodec.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libavformat.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libavutil.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libswresample.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libhss1394.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/liblilv.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libsord.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libserd.a  lib/libshout-idjc/libshout_mixxx.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libogg.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libvorbis.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libvorbisenc.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libopusfile.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libopus.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libmodplug.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libqt5keychain.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5Core.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libsqlite3.a  -weak_framework AppKit  -weak_framework CoreFoundation  -weak_framework CoreServices  -weak_framework CoreVideo  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libjpeg.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libfreetype.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpng16.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libbz2.a  -Xlinker -framework -Xlinker ApplicationServices  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libfreetype.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libz.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libbz2.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpng16.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libz.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libbrotlidec-static.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libbrotlicommon-static.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libdouble-conversion.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpq.a  -ldl  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libssl.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libcrypto.a  -weak_framework DiskArbitration  -weak_framework IOKit  -weak_framework Foundation  -weak_framework Security  -weak_framework ApplicationServices  -weak_framework SystemConfiguration  -weak_framework Carbon  -weak_framework QuartzCore  -weak_framework Metal  -weak_framework CoreText  -weak_framework CoreGraphics  -weak_framework OpenGL  -weak_framework AGL  -weak_framework ImageIO  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpcre2-16.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5ThemeSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5EventDispatcherSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5PlatformCompositorSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5FontDatabaseSupport.a  -lz  -lm  -lcups  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5GraphicsSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5ClipboardSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5AccessibilitySupport.a  -lqcocoa_LIBRARY_DEBUG-NOTFOUND  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpcre2-16.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5ThemeSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5EventDispatcherSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5PlatformCompositorSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5FontDatabaseSupport.a  -lz  -lm  -lcups  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5GraphicsSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5ClipboardSupport.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5AccessibilitySupport.a  -lqcocoa_LIBRARY_DEBUG-NOTFOUND  -framework Foundation  -framework Security  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libhidapi.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libusb-1.0.a  libmixxx-xwax.a  /Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libwavpack.a && :
ld: warning: ignoring duplicate libraries: '-lcups', '-lm', '-lqcocoa_LIBRARY_DEBUG-NOTFOUND', '-lz', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5AccessibilitySupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5ClipboardSupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5EventDispatcherSupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5FontDatabaseSupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5GraphicsSupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5PlatformCompositorSupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libQt5ThemeSupport.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libbz2.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libfreetype.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpcre2-16.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libpng16.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libsqlite3.a', '/Users/<user>/git/mixxx/buildenv/mixxx-deps-rel-2.4-x64-osx-min1012-release-c82358c/installed/x64-osx-min1012-release/lib/libz.a'
ld: library 'qcocoa_LIBRARY_DEBUG-NOTFOUND' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[787/885] Building CXX object CMakeFiles/mixxx-test.dir/src/test/autodjprocessor_test.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by fwcd** [comment]
Unfortunately, I am still getting the same error after applying the patch.

**Comment by JoergAtGithub** [comment]
@daschuer Try to rename it from .bat to .cmd
https://stackoverflow.com/questions/71477056/bat-and-cmd-different-on-errorlevel-who-determines-sets-behavior

**Comment by JoergAtGithub** [comment]
I tested this locally on Windows11 and it still works as expected.
But I noticed, that our buildenvs are missing the Qt PDB files for RelWithDebugInfo the DLLs - but this is not a regression of this PR.

**Comment by daschuer** [comment]
Ok, this is fine, finally. 

The missing PDB is kind of expected, because vcpkg has no RelWithDebInfo and user the release binary without debug information.

**Comment by JoergAtGithub** [comment]
But all other VCPKG DLLs have PDB files - only for Qt (where it's most important) they are missing. Also our 2.5 buildenv comes with PDB files for the Qt DLLs.

**Comment by daschuer** [comment]
Here is a PR the brings PDB files in Release mode: https://github.com/mixxxdj/vcpkg/pull/109

**Comment by daschuer** [comment]
All green. Merge?

**Comment by JoergAtGithub** [comment]
I'm no sure if the local Debug build issue for macOS is fixed, but since this wouldn't be a regression, lets merge it! Thank you very much!

**Pull request merged**